### ISSUE TITLE
Omitting tag during sync results in an error when using skopeo, works using docker

### ISSUE THREAD

**Issue by gleb-zverinskiy** [xelalexv/dregsy#13]
The documentation states that if the tag is omitted all tags will be synced.

I am syncing from aws ecr to artifactory, when I provide a tag the sync works, when I don't it does not.

working:
```    # 'mappings' is a list of 'from':'to' pairs that define mappings of image
    # paths in the source registry to paths in the destination; 'from' is
    # required, while 'to' can be dropped if the path should remain the same as
    # 'from'. Additionally, the tags being synced for a mapping can be limited
    # by providing a 'tags' list. When omitted, all image tags are synced.
    mappings:
      - from: dev-image1
        to: dev-image1
        tags: ['0snapshot-dev']
#      - from: test/another-image
```
``` 
docker run --rm -v ./config.yaml:/config.yaml  xelalex/dregsy
2020-05-08T15:37:34Z [INFO] 
2020-05-08T15:37:34Z [INFO] skopeo version 0.1.32
2020-05-08T15:37:34Z [INFO] skopeo relay ready
2020-05-08T15:37:34Z [INFO] syncing task 'task1': 'ecr.aws.com' --> 'destination.com'
2020-05-08T15:37:34Z [INFO] mapping '/dev-image1' to '/dev-image1'
2020-05-08T15:37:34Z [INFO] 
2020-05-08T15:37:34Z [INFO] syncing tag '0snapshot-dev':
Getting image source signatures
Skipping fetch of repeat blob sha256:123275d6e508d282237a22fefa5aef822b719a06496444ea89efa65da523fc4b
Skipping fetch of repeat blob sha256:6cd6a943ce277e76a108b1c860c5991d3c9acae2db14865c7a6d4a2231b38ff2
Skipping fetch of repeat blob sha256:a50b5ac4a7fb2ab0d0cc3f2d3c1063d9f7751e32fe038b7ad7545bb239027167
Skipping fetch of repeat blob sha256:a2713a4ab6152352dc7660cb029cda5ba26bb7d7dbeb80ff38678f052dac97e5
Skipping fetch of repeat blob sha256:2db755cb04039e83097e96b20cc3bffb5752498fa1e0105356447b2d9df4f0fd
Skipping fetch of repeat blob sha256:a7febb4cdc5f9be8051af037bf73b9a69644283b223b6866504d67d025a91d5d
Copying config sha256:20d5fe6d7deec98a85813f251ec3264505c4bc6fbc8dda7061c0a3c0a81fed1e
 0 B / 7.22 KB  0s
Writing manifest to image destination
Storing signatures
2020-05-08T15:37:47Z [INFO] 
2020-05-08T15:37:47Z [INFO] all done
```

not working:
```    
# 'mappings' is a list of 'from':'to' pairs that define mappings of image
    # paths in the source registry to paths in the destination; 'from' is
    # required, while 'to' can be dropped if the path should remain the same as
    # 'from'. Additionally, the tags being synced for a mapping can be limited
    # by providing a 'tags' list. When omitted, all image tags are synced.
    mappings:
      - from: dev-image1
        to: dev-image1
#        tags: ['0snapshot-dev']
#      - from: test/another-image
```
```
docker run --rm -v .config.yaml:/config.yaml  xelalex/dregsy
2020-05-08T15:43:11Z [INFO] 
2020-05-08T15:43:11Z [INFO] skopeo version 0.1.32
2020-05-08T15:43:11Z [INFO] skopeo relay ready
2020-05-08T15:43:11Z [INFO] syncing task 'task1': 'ecr.aws.com' --> 'destination.com'
2020-05-08T15:43:11Z [INFO] mapping '/dev-image1' to '/dev-image1'
2020-05-08T15:43:12Z [INFO] 
2020-05-08T15:43:12Z [INFO] all done
2020-05-08T15:43:12Z [ERROR] error listing image tags: time="2020-05-08T15:43:12Z" level=fatal msg="Error reading manifest latest in ecr.aws.com/dev-image1: manifest unknown: Requested image not found" 
, exit status 1
```

**Comment by xelalexv** [comment]
Another thought, if you're using an instance profile, the permissions listed in the README may actually not be sufficient. You could try with this ECR Read-Only Policy from AWS:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "ecr:GetAuthorizationToken",
                "ecr:BatchCheckLayerAvailability",
                "ecr:GetDownloadUrlForLayer",
                "ecr:GetRepositoryPolicy",
                "ecr:DescribeRepositories",
                "ecr:ListImages",
                "ecr:DescribeImages",
                "ecr:BatchGetImage",
                "ecr:GetLifecyclePolicy",
                "ecr:GetLifecyclePolicyPreview",
                "ecr:ListTagsForResource",
                "ecr:DescribeImageScanFindings"
            ],
            "Resource": "*"
        }
    ]
}
```
 More here: https://docs.aws.amazon.com/AmazonECR/latest/userguide/ecr_managed_policies.html#AmazonEC2ContainerRegistryReadOnly

**Comment by aceresia-bg** [comment]
I'm having the same issue.  The problem is that I didn't have an image with the "latest" tag.

**Comment by xelalexv** [comment]
Could you run `skopeo inspect` for your situation as per my comment further up?

**Comment by aceresia-bg** [comment]
FATA[0000] Error parsing image name "docker://<image location>": Error reading manifest latest in <image location>: manifest unknown: The named manifest is not known to the registry.

the image location didn't contain a tag, and there is no latest tag.  If I create a latest tag, the sync works again.

**Comment by xelalexv** [comment]
OK, this is caused by [this issue](https://github.com/containers/skopeo/issues/705) in Skopeo. `skopeo inspect` is not really meant for listing tags, that's just a side-effect, but until recently, there was no dedicated command for that. Now, [ther is one](https://github.com/containers/skopeo/pull/716). However, that's only available starting with v0.2.0, so I'll have to bump the Skopeo version used with Dregsy.

**Comment by xelalexv** [comment]
I upgraded to Skopeo 0.2.0, and switched to using Skopeo's `list-tags` command and did some testing. Seems fine, but wouldn't hurt if you could try this out with your particular cases (see branch `issue13`, I also made this available as Docker image `xelalex/dregsy:issue13`).

**Comment by fredgate** [comment]
I have a repository with no `latest` tag, and with dregsy `0.3.0` I had the error.  
I tested dregsy image `issue13` and it now works fine. This a great improvment :-)

**Comment by xelalexv** [comment]
Great, and thanks for testing! I'll merge this then.

**Comment by xelalexv** [comment]
fixed with pr #17
